gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -0,0 +1,480 @@
+  }
+
+  @Test
+  public void testGetProjects() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.getProjects().test().assertValue(ImmutableList.of(TEST_PROJECT));
+  }
+
+  @Test
+  public void testGetProjectById() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.getProjectById("project id").test().assertValue(TEST_PROJECT);
+  }
+
+  @Test
+  public void testDeleteProject() {
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+    localDataStore.deleteProject(TEST_PROJECT).test().assertComplete();
+    localDataStore.getProjects().test().assertValue(AbstractCollection::isEmpty);
+  }
+
+  @Test
+  public void testInsertUser() {
+    localDataStore.insertOrUpdateUser(TEST_USER).test().assertComplete();
+  }
+
+  @Test
+  public void testGetUser() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.getUser("user id").test().assertValue(TEST_USER);
+  }
+
+  @Test
+  public void testApplyAndEnqueue_featureMutation() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+
+    localDataStore.applyAndEnqueue(TEST_FEATURE_MUTATION).test().assertComplete();
+
+    // assert that mutation is saved to local database
+    localDataStore
+        .getPendingMutations("feature id")
+        .test()
+        .assertValue(ImmutableList.of(TEST_FEATURE_MUTATION));
+
+    // assert feature is saved to local database
+    Feature feature = localDataStore.getFeature(TEST_PROJECT, "feature id").blockingGet();
+    assertFeature("feature id", TEST_POINT, feature);
+  }
+
+  @Test
+  public void testGetFeaturesOnceAndStream() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+
+    TestSubscriber<ImmutableSet<Feature>> subscriber =
+        localDataStore.getFeaturesOnceAndStream(TEST_PROJECT).test();
+
+    subscriber.assertValueCount(1);
+    subscriber.assertValueAt(0, AbstractCollection::isEmpty);
+
+    FeatureMutation mutation = TEST_FEATURE_MUTATION;
+    localDataStore.applyAndEnqueue(mutation).blockingAwait();
+
+    Feature feature =
+        localDataStore.getFeature(TEST_PROJECT, mutation.getFeatureId()).blockingGet();
+
+    subscriber.assertValueCount(2);
+    subscriber.assertValueAt(0, AbstractCollection::isEmpty);
+    subscriber.assertValueAt(1, ImmutableSet.of(feature));
+  }
+
+  @Test
+  public void testUpdateMutations() {
+    localDataStore.insertOrUpdateUser(TEST_USER).blockingAwait();
+    localDataStore.insertOrUpdateProject(TEST_PROJECT).blockingAwait();
+
+    FeatureMutation mutation = TEST_FEATURE_MUTATION;
+    localDataStore.applyAndEnqueue(mutation).blockingAwait();
+
+    Point newPoint = Point.newBuilder().setLatitude(51.0).setLongitude(44.0).build();
```

Comment:
Shall we create a constant, TEST_POINT_2, for example, for use here and throughout instead? Otherwise it may seem like these values might have special meaning.

Also, consider creating a constructor method createTestFeatureMutation(point) that you can use for both cases.